python/cudf/cudf/core/column/column.py

```diff
@@ -854,12 +854,16 @@ def as_categorical_column(self, dtype, **kwargs):
                 ordered=ordered,
             ).astype(dtype)
 
-        labels, cats = sr.factorize()
```

Comment:
Should factorize just be changed to unsigned types?

Reply:
We change it, but it might diverge from pandas and we will not be able to use `-ve` sentinel values.

Reply:
Lets leave this as is for now